lib/paloma/action_controller_extension.rb

```diff
@@ -107,47 +107,24 @@ def track_paloma_request
 
 
       #
-      # Before rendering html reponses,
-      # this is exectued to append Paloma's html hook to the response.
+      # Call in your view to insert Paloma's html hook.
       #
       # The html hook contains the javascript code that
       # will execute the tracked Paloma requests.
       #
-      def append_paloma_hook
+      def insert_paloma_hook
         return true if self.paloma.has_no_request?
-       
-        # Render the partial if it is present, otherwise do nothing. 
-        begin
-          hook = view_context.render(
-                   :partial => 'paloma/hook',
-                   :locals => {:request => self.paloma.request})
-        rescue ActionView::MissingTemplate
-          return true
-        end
-
-        before_body_end_index = response_body[0].rindex('</body>')
 
-        # Append the hook after the body tag if it is present.
-        if before_body_end_index.present?
-          before_body = response_body[0][0, before_body_end_index].html_safe
-          after_body = response_body[0][before_body_end_index..-1].html_safe
-
-          response.body = before_body + hook + after_body
-        else
-          # If body tag is not present, append hook in the response body
-          response.body += hook
-        end
+        hook = view_context.render(
+                 :partial => 'paloma/hook',
```

Comment:
Indent the first parameter one step more than the previous line.<br>Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.